internal/tui/portforward_handlers.go

```diff
@@ -1,175 +1,234 @@
+	if wcName != "" {
+		m.portForwardOrder = append(m.portForwardOrder, wcPaneFocusKey)
 	}
-	return m, nil
-}
 
-func handlePortForwardOutputMsg(m model, msg portForwardOutputMsg) (model, tea.Cmd) {
-	var cmds []tea.Cmd
-	if pf, ok := m.portForwards[msg.label]; ok {
-		line := fmt.Sprintf("[%s %s] %s", msg.label, msg.streamType, msg.line)
-		pf.output = append(pf.output, msg.line)
-		m.combinedOutput = append(m.combinedOutput, line)
-
-		if !pf.forwardingEstablished && strings.Contains(msg.line, "Forwarding from") {
-			pf.forwardingEstablished = true
-			// Try to extract the local port more robustly
-			parts := strings.Fields(msg.line)
-			localPort := "unknown"
-			for i, p := range parts {
-				if (p == "from" || p == "From") && i+1 < len(parts) {
-					addressAndPort := parts[i+1]
-					lastColon := strings.LastIndex(addressAndPort, ":")
-					if lastColon != -1 && lastColon+1 < len(addressAndPort) {
-						localPort = addressAndPort[lastColon+1:]
-						break
-					}
-				}
-			}
-			pf.statusMsg = fmt.Sprintf("Forwarding Active (Local Port: %s)", localPort)
-			m.combinedOutput = append(m.combinedOutput, fmt.Sprintf("[%s] Confirmed: %s", msg.label, pf.statusMsg))
+	// Prometheus for MC
+	if mcName != "" {
+		promLabel := "Prometheus (MC)"
+		m.portForwardOrder = append(m.portForwardOrder, promLabel)
+		m.portForwards[promLabel] = &portForwardProcess{
+			label:     promLabel,
+			port:      "8080:8080",
+			isWC:      false,
+			context:   "teleport.giantswarm.io-" + mcName, // mcName is sufficient, no need for m.getManagementClusterContextIdentifier()
+			namespace: "mimir",
+			service:   "service/mimir-query-frontend",
+			active:    true,
+			statusMsg: "Awaiting Setup...",
 		}
 
-		// Trim outputs
-		if len(m.combinedOutput) > maxCombinedOutputLines {
-			m.combinedOutput = m.combinedOutput[len(m.combinedOutput)-maxCombinedOutputLines:]
-		}
-		if len(pf.output) > maxCombinedOutputLines { // Assuming same trim size for individual PFs
-			pf.output = pf.output[len(pf.output)-maxCombinedOutputLines:]
+		// Grafana for MC
+		grafanaLabel := "Grafana (MC)"
+		m.portForwardOrder = append(m.portForwardOrder, grafanaLabel)
+		m.portForwards[grafanaLabel] = &portForwardProcess{
+			label:     grafanaLabel,
+			port:      "3000:3000",
+			isWC:      false,
+			context:   "teleport.giantswarm.io-" + mcName, // mcName is sufficient
+			namespace: "monitoring",
+			service:   "service/grafana",
+			active:    true,
+			statusMsg: "Awaiting Setup...",
 		}
+	}
 
-		// Continue listening on streams if they are not closed
-		if msg.streamType == "stdout" && !pf.stdoutClosed {
-			cmds = append(cmds, waitForPortForwardActivity(msg.label, "stdout", pf.stdout))
-		} else if msg.streamType == "stderr" && !pf.stderrClosed {
-			cmds = append(cmds, waitForPortForwardActivity(msg.label, "stderr", pf.stderr))
+	// Alloy Metrics for WC
+	if wcName != "" {
+		alloyLabel := "Alloy Metrics (WC)"
```

Comment:
Alloy is also on MC so this should not be limited to WC

Reply:
Thanks! Implemented.